lib/compute/instance-group.js

```diff
@@ -0,0 +1,471 @@
+        pageToken: resp.nextPageToken
+      });
+    }
+
+    var vms = arrify(resp.items).map(function(vm) {
+      var vmInstance = self.zone.vm(vm.instance);
+      vmInstance.metadata = vm;
+      return vmInstance;
+    });
+
+    callback(null, vms, nextQuery, resp);
+  });
+};
+
+/**
+ * Remove one or more VMs from this instance group.
+ *
+ * @resource [InstanceGroups: removeInstances API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroups/removeInstances}
+ *
+ * @param {module:compute/vm|module:compute/vm[]} vms - VM instances to remove
+ *     from this instance group.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:compute/operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var vms = [
+ *   gce.zone('us-central1-a').vm('http-server'),
+ *   gce.zone('us-central1-a').vm('https-server')
+ * ];
+ *
+ * instanceGroup.remove(vms, function(err, operation, apiResponse) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ */
+InstanceGroup.prototype.remove = function(vms, callback) {
+  var self = this;
+
+  this.request({
+    method: 'POST',
+    uri: '/removeInstances',
+    json: {
+      instances: arrify(vms).map(function(vm) {
+        return {
+          instance: vm.url
+        };
+      })
+    }
+  }, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var operation = self.zone.operation(resp.name);
+    operation.metadata = resp;
+
+    callback(err, operation, resp);
+  });
+};
+
+/**
+ * Set the named ports for this instance group.
+ *
+ * @resource [InstanceGroups: setNamedPorts API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroups/setNamedPorts}
+ *
+ * @param {object} ports - A map of names to ports. The key should be the name,
+ *     and the value the port number.
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:compute/operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var ports = {
+ *   http: 80,
+ *   https: 443
+ * };
+ *
+ * instanceGroup.setPorts(ports, function(err, operation, apiResponse) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ */
+InstanceGroup.prototype.setPorts = function(ports, callback) {
+  var self = this;
+
+  callback = callback || util.noop;
+
+  var namedPorts = [];
+
+  for (var namedPort in ports) {
```

Comment:
Any reason we don't do an `Object.keys(..).map(..)` here? This is a super-nitpick, so feel free to ignore.. but I think historically `for..in` loops have been uncommon for our codebase, so it made me raise an eyebrow.